src/main/kotlin/de/klg71/keycloakmigration/changeControl/MigrationChangelog.kt

```diff
@@ -110,7 +110,7 @@ internal class MigrationChangelog(private val migrationUserId: UUID, private val
 
     private fun List<String>.replaceString(oldHash: String, newHash: String): List<String> {
         return map {
-            if (it == oldHash) {
+            if (it.contains(oldHash)) {
```

Comment:
I wouldn't like to see this change here. 
I have an idea where this might come from:
I designed the migration from "v1" to "v2" in a way that it is currently not possible to migrate if you hashes are not correct.
So you can only do one thing migrate or correct all hashes.
I guess we could also add an option to just rewrite all hashes.

But if you have another error here maybe you could give me the configuration, so I can reconstruct it in my debugger :)

Reply:
I thought this functionality is a way to "correct hashes" :D.

I needed to correct hashes because my old changes aren't working in clean environment with new keycloak 20.0.3. Previously these changes worked in clean environment with keycloak 14.

So I updated my old change yaml and of course hashes are now different for existing environments, for example, in production env.

I used --correct-hashes option and it worked for existing environments. It was updating hashes and consequent keycloak migration calls are finishing without hash mismatch errors.

Later from testing we figured out that hashes are only updated when I run keycloak migration with --correct-hashes and also with some unexecuted changes. If there was no change to execute - all migration are skipped - then hash replacements are announced in logs but aren't actually replaced in keycloak. So in production environment case I used those logs with replacement instructions and performed replacement in user migration attributes myself.

I don't think that we should create a specific functionality for this case because it isn't recommended and this situation doesn't happen often.

Reply:
Ok I guess I see the problem. I think I could try to add this edge case to the documentation.
I will try to write up something from your comment here. If you have any other documentation for the process you used available, I think it could be useful :)

So what do we do with the `.contains`-Call ^^. In my eyes there should only be hash values submitted to this function. Without the whole `v2/{index}`-stuff. And if this is the case the `==` operator should be sufficient.

Reply:
yes, only one type of data - hash - should be in this function. Then we could only use == operator. Can you update and test this?

Reply:
I'll try to make this change

Reply:
@klg71 I made changes. please check whether it is ok